File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        result = [int(digit) for digit in str(n)]

        sum_n = 0
        prod_n = 1

        for i in range(len(result)):
            sum_n += result[i]
            prod_n *= result[i]

        if (sum_n + prod_n) == 0:
            return False

        if n % (sum_n + prod_n) == 0:
            return True
        else:
            return False

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int s = 0, p = 1;
        int x = n;
        
        while (x != 0) {
            int v = x % 10; // Extract the last digit
            x /= 10;        // Remove the last digit
            s += v;         // Add to digit sum
            p *= v;         // Multiply with digit product
        }
        
        // Check if n is divisible by the sum of digit sum and digit product
        return n % (s + p) == 0;
    }
}